rustls/src/crypto/signer.rs

```diff
@@ -123,6 +125,29 @@ pub struct CertifiedKey {
 }
 
 impl CertifiedKey {
+    /// Create a new `CertifiedKey` from a certificate chain and DER-encoded private key.
+    ///
+    /// Attempt to parse the private key with the given [`CryptoProvider`]'s [`KeyProvider`] and
+    /// verify that it matches the public key in the first certificate of the `cert_chain`.
```

Comment:
nit: maybe add "if possible" here so that the way this code eats `Error::InconsistentKeys(InconsistentKeys::Unknown)` when `keys_match` isn't wired up doesn't come as a surprise?

Reply:
Makes sense, done.